### PR TITLE
fix(make help, detect-secrets): adding detect-secrets with python, `make help`

### PR DESCRIPTION
**Pull request by pvinis**
The type of this PR is: **Enhancement**

This PR resolves [CX-1799]

### Description

In this PR we have a few things:
- We figured out why the detect-secrets was failing, and tried to fix it.
- We add a `make help` command that runs the previously named `diagnose` script by @MounirDhahri.
- We add new checks on `make help`, some basic tools ones and some for detect-secrets.
- Also a quick upgrade of husky 4->7 and node (minor).
- We add `make install` to run all the installers, `bundle install`, `yarn install`, `pod install`, `pip install`, all in one.
- We stop `pod install` from running `yarn install`. We can keep them combined with `make install`.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- a bunch of tooling work for `make help` and python - pavlos
- making detect-secrets work - ole, pavlos

</details>

[CX-1799]: https://artsyproduct.atlassian.net/browse/CX-1799

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (a bunch of tooling work for `make help` and python - pavlos,making detect-secrets work - ole, pavlos)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e5405655098044fcbb2841c5128215bc014e635c
</p>

**Review by olerichter00 — APPROVED** [review]
Great work!

File: package.json [hunk 1]
```diff
@@ -293,13 +291,6 @@
     "trailingComma": "es5",
     "bracketSpacing": true
   },
-  "husky": {
-    "hooks": {
-      "post-merge": "pull-lock",
-      "pre-commit": "lint-staged",
-      "pre-push": "yarn type-check"
-    }
-  },
```
[the remark this answers is not in the document]
Comment (pvinis) [reply package.json:302]: I upgraded us to hustky 7, and that's the way they do it. it also makes it easier I guess for them to just make them got hook actually? just the preferred way they suggested, and I like simplifying package.json, so here we are 😀

**Comment by pvinis** [comment]
I'll fiz the podfile soon

**Comment by olerichter00** [comment]
#squashongreen

**Pull request merged**